### PR TITLE
Ldap healthcheck

### LINKED ISSUE #367: Add LDAP healthcheck in the portal
see https://github.com/vinyldns/vinyldns/issues/357

validate connectivity to LDAP from the portal. do test user get, not found or found is ok, just make sure it's not timing out.

### PR DESCRIPTION
**Pull request by rebstar6**
Fixes #367

### CODE REVIEW

File: modules/portal/app/controllers/LdapAuthenticator.scala [hunk 1]
```diff
@@ -198,11 +201,28 @@ class LdapAuthenticator(
   def lookup(username: String): Try[UserDetails] =
     findUserDetails(searchBase, username, authenticator.lookup(_, username, serviceAccount))
 
+  def checkHealth(): HealthCheck =
+    IO {
```
Comment (rebstar6) [inline modules/portal/app/controllers/LdapAuthenticator.scala]: this is gross. i need to figure out some better scala for this (suggestions welcome).

basically, `authenticator.lookup(domain, "healthlookup", serviceAccount)` is a Try. In the case that the Try is successful, or it's a UserDoesNotExistException, we are fine. Any other error means failure.

Comment (mitruly) [reply modules/portal/app/controllers/LdapAuthenticator.scala]: This makes sense; not sure if there is a more concise way to convey this; playing locally, the following worked (but is functionally the same):

```
  def healthCheck(): HealthCheck =
    IO {
      searchBase.headOption
        .map { domain =>
          authenticator.lookup(domain, "healthlookup", serviceAccount).as(()).recover {
            case _: UserDoesNotExistException => ()
          }
        } match {
        case Some(Failure(exception)) => exception.asLeft
        case _ => ().asRight
      }
    }.asHealthCheck
```

Comment (rebstar6) [reply modules/portal/app/controllers/LdapAuthenticator.scala]: re-pushed more similar to what you have. thoughts @mitruly?

Comment (mitruly) [reply modules/portal/app/controllers/LdapAuthenticator.scala]: Me likey

File: modules/portal/app/controllers/LdapAuthenticator.scala [hunk 2]
```diff
@@ -185,9 +188,9 @@ class LdapAuthenticator(
     extends Authenticator {
 
   private def findUserDetails(
-      domains: List[LdapSearchDomain],
-      userName: String,
-      f: LdapSearchDomain => Try[UserDetails]): Try[UserDetails] = domains match {
+                               domains: List[LdapSearchDomain],
```
Comment (mitruly) [inline modules/portal/app/controllers/LdapAuthenticator.scala]: Looks like some weird IntelliJ formatting.

Comment (rebstar6) [reply modules/portal/app/controllers/LdapAuthenticator.scala]: i hate when it does this.

**Comment by britneywright** [comment]
this is an edge case, but if you run the portal locally without using LDAP the health endpoint errors. Do we want to handle that or at least make note of it somewhere?

**Comment by rebstar6** [comment]
hmm a good question @britneywright. im inclined to take the dummy ldap domain out of reference.conf now that I look at it. will dig.

**Pull request merged**